sentry/src/main/java/io/sentry/Scope.java

```diff
@@ -140,7 +144,13 @@ public ISpan getSpan() {
    * @param transaction the transaction
    */
   public void setTransaction(final @NotNull ITransaction transaction) {
-    this.transaction = Objects.requireNonNull(transaction, "transaction is required");
+    synchronized (transactionLock) {
+      this.transaction = Objects.requireNonNull(transaction, "transaction is required");
+    }
+
+    // TODO: double-check this
+    // setTransaction(String) replaces the transaction.name so I guess this makes sense too
```

Comment:
@bruno-garcia @maciejwalkowiak opinions?

Reply:
When we phase out `Transaction` being on the scope this would introduce a breaking change since creating a root span with `startTransaction` would now require also changing the scope's `transaction` property with the given string. No strong opinions here but this seems like a can of warms.

Reply:
I will keep as it is then